core/main.go

```diff
@@ -40,9 +41,20 @@ func Run(cfg *Config) {
 
 }
 
+func addDefaultFilteringMode(cfg *Config) {
+	if cfg.TagFilteringMode == "" {
+		cfg.TagFilteringMode = "opt-in"
+	}
```

Comment:
I would change that method to make it as such:
```go
if cfg.TagFilteringMode != "opt-out" {
```

This is because there are two valid options: `opt-in` (default) and `opt-out`. But with the current method, the validation of `cfg.TagFilteringMode = "whatever"` would result in weird behaviour; while with this change anything not being valid would result in `opt-in`. You could also add a debug/warning message for the users; or even stop the execution, but that's a bit harsh.